src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDetailsMapping.java

```diff
@@ -83,19 +80,54 @@ public static HearingDetails buildHearingDetails(HearingWrapper wrapper,
             .panelRequirements(getPanelRequirements(caseData, referenceDataServiceHolder))
             .hearingIsLinkedFlag(isCaseLinked(caseData))
             .amendReasonCode(getAmendReasonCode())
+            .hearingChannels(getHearingChannel(caseData))
             .build();
     }
 
+    static List<HearingChannel> getHearingChannel(SscsCaseData caseData) {
+        if (caseData.getDwpIsOfficerAttending() != null && caseData.getDwpIsOfficerAttending().equals(YES.getValue())) {
+            return Collections.singletonList(FACE_TO_FACE);
+        }
+
+        List<HearingChannel> hearingChannels = new ArrayList<>();
+
+        HearingChannel individualPreferredHearingChannel = getIndividualPreferredHearingChannel(
+            caseData.getAppeal().getHearingSubtype(),
+            caseData.getAppeal().getHearingOptions()
+        );
+        hearingChannels.add(individualPreferredHearingChannel);
+
+        if (nonNull(caseData.getOtherParties())) {
+            hearingChannels.addAll(caseData.getOtherParties().stream().map(CcdValue::getValue).map(
+                    otherParty -> getIndividualPreferredHearingChannel(
+                        otherParty.getHearingSubtype(),
+                        otherParty.getHearingOptions()
+                    ))
+                                       .collect(Collectors.toList()));
+        }
+
+        if (hearingChannels.contains(FACE_TO_FACE)) {
+            return Collections.singletonList(FACE_TO_FACE);
+        } else if (hearingChannels.contains(VIDEO)) {
+            return Collections.singletonList(VIDEO);
+        } else if (hearingChannels.contains(TELEPHONE)) {
+            return Collections.singletonList(TELEPHONE);
+        } else {
+            return Collections.singletonList(NOT_ATTENDING);
+        }
+    }
+
+
     public static String getHearingType() {
         return SUBSTANTIVE.getHmcReference();
     }
 
     public static HearingWindow buildHearingWindow(@Valid SscsCaseData caseData, boolean autoListed) {
         return HearingWindow.builder()
-                .firstDateTimeMustBe(getFirstDateTimeMustBe())
-                .dateRangeStart(getHearingWindowStart(caseData, autoListed))
-                .dateRangeEnd(null)
-                .build();
+            .firstDateTimeMustBe(getFirstDateTimeMustBe())
+            .dateRangeStart(getHearingWindowStart(caseData, autoListed))
+            .dateRangeEnd(null)
+            .build();
```

Comment:
Can you avoid changing the formatting of an entire class, it makes it hard to review